ddtrace/ext/ci.py

```diff
@@ -41,28 +44,37 @@
 
 
 def _normalize_ref(name):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_TAGS.sub("", _RE_ORIGIN.sub("", _RE_REFS.sub("", name))) if name is not None else None
 
 
 def _filter_sensitive_info(url):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_URL.sub("\\1", url) if url is not None else None
 
 
 def tags(env=None):
+    # type: (Optional[MutableMapping[str, str]]) -> Dict[str, str]
+    """Extract and set tags from provider environ, as well as git metadata."""
     env = os.environ if env is None else env
```

Comment:
you might be able to do `  # type: Dict[str, str]` here if you want to completely avoid `MutableMapping`, but this isn't a big deal.

Reply:
Unfortunately in line 59 we default to `os.environ` which isn't a `Dict[str, str]` and is instead a `_Environ[str]`, which are apparently both a subclass of `MutableMapping` and so this might have to stay 😢

Reply:
all good, this is also me trying to better understand `MutableMapping` when/where/how we want to be using it in our code base as well.